decksite/data/rule.py

```diff
@@ -0,0 +1,189 @@
+from typing import List
+
+from decksite.data import deck
+from decksite.database import db
+from magic.models import Deck
+from shared.container import Container
+
+
+def num_classified_decks() -> int:
+    sql = 'SELECT COUNT(DISTINCT(deck_id)) AS c FROM ({apply_rules_query}) AS applied_rules'.format(apply_rules_query=apply_rules_query())
+    return (db().select(sql))[0]['c']
```

Comment:
Same comments here about `DISTINCT` and about `db().value(sql)`

Reply:
The DISTINCT isn't redundant here if we're in the situation where we have a deck which currently fits multiple rules, right?

Reply:
Oh, this is more complicated than I thought, then :D

Reply:
The result set of the SQL contains one row for each `(deck_id, rule_id)` pair that could apply (it also contains the `archetype_id` for convenience, but that information can be extracted from the `rule_id`).